apps/calculation/calculation.cpp

```diff
@@ -156,11 +160,18 @@ KDCoordinate Calculation::height(Context * context, bool expanded, bool allExpre
 
   if (displayOutput(context) == DisplayOutput::ExactOnly) {
     KDCoordinate exactOutputHeight = exactLayout.layoutSize().height();
-    if (allExpressionsInline) {
+    KDCoordinate exactOutputWidth = exactLayout.layoutSize().width();
+    singleLine = exactOutputWidth + inputWidth < maxWidth - 40 - 37;
```

Comment:
Could you express these magic numbers in terms of other object widths? I haven't investigated which ones but maybe: Metric::EllipsisCellWidth, Metric::CommonSmallMargin?

Reply:
I tried to do it with Metric::EllipsisCellWidth and the text would go invisible/ be hidden behind the ellipsis whenever it was selected.

Reply:
I changed "-37" to Metric::EllipsisCellWidth in my latest commit. the "-40" or "-70" is still required though